beetsplug/spotify.py

```diff
@@ -536,3 +556,71 @@ def _output_match_results(self, results):
             self._log.warning(
                 f'No {self.data_source} tracks found from beets query'
             )
+
+    def _fetch_info(self, items, write, force):
+        """Obtain track information from Spotify."""
+        spotify_audio_features = {
+            'acousticness': ['spotify_track_acousticness'],
+            'danceability': ['spotify_track_danceability'],
+            'energy': ['spotify_track_energy'],
+            'instrumentalness': ['spotify_track_instrumentalness'],
+            'key': ['spotify_track_key'],
+            'liveness': ['spotify_track_liveness'],
+            'loudness': ['spotify_track_loudness'],
+            'mode': ['spotify_track_mode'],
+            'speechiness': ['spotify_track_speechiness'],
+            'tempo': ['spotify_track_tempo'],
+            'time_signature': ['spotify_track_time_sig'],
+            'valence': ['spotify_track_valence'],
+        }
+        import time
+
+        no_items = len(items)
+        self._log.info('Total {} tracks', no_items)
+
+        for index, item in enumerate(items, start=1):
+            time.sleep(.5)
```

Comment:
To avoid [API rate limit](https://developer.spotify.com/documentation/web-api/guides/rate-limits/).

Reply:
I think it would be great to have a code comment that explains this. Would it even be possible to check for a 429 response code in `SpotifyPlugin._handle_response`, and retry after the provided `Retry-After` value (and maybe default to some exponential backoff, the documentation doesn't seem to guarantee that this header is always pressent)?

Reply:
@wisp3rwind I agree it would be great, but I am not sure about the implementation at this point. I have tried not to touch the `_handle_response` function for precisely that reason. Once this PR is merged, hopefully, we can add that feature to make the plugin more robust. With the `sleep` and if someone is using their own keys, we should not hit 429.

Reply:
Meanwhile, I have added a comment to reflect that.

Reply:
Fair enough, let's open a new issue to remember the issue of rate limiting.

Reply:
Created an issue https://github.com/beetbox/beets/issues/4369 to keep a track of it.